src/crimson/os/seastore/omap_manager/btree/omap_btree_node.h

```diff
@@ -0,0 +1,142 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <boost/iterator/counting_iterator.hpp>
+
+#include "crimson/common/log.h"
+#include "crimson/os/seastore/seastore_types.h"
+#include "crimson/os/seastore/transaction_manager.h"
+#include "crimson/os/seastore/omap_manager.h"
+#include "crimson/os/seastore/omap_manager/btree/omap_types.h"
+
+namespace crimson::os::seastore::omap_manager{
+
+struct omap_context_t {
+  omap_root_t &omap_root;
+  TransactionManager &tm;
+  Transaction &t;
+};
+
+struct OMapNode : LogicalCachedExtent {
+  using OMapNodeRef = TCachedExtentRef<OMapNode>;
+  OMapNodeRef parent = nullptr;
```

Comment:
What's this for?

Reply:
Ah, it's how you are finding the parent objects in checking_parent.  This won't work.  This reference will keep *a* copy of the parent instance in memory, but as soon as another mutation causes that node to be duplicated for write, the pointers will point at some now obsolete copy that doesn't reflect the current state of the object.  We can discuss this more next week, but as I indicated in another comment, the problem at its core is that you need to handle splits and merges starting at the leaf as each mutation returns.